### PR TITLE
enhancement: spawn location

### PR DESCRIPTION
**Pull request by whitew1994WW**
Ok, I have redone the spawning. Now you have the option of either spawn lines or spawn points. You also have the option of choosing the enemy that will spawn on that point or line by adding a child node of the enemy to spawn along it.

I have re-adjusted the way the room difficulty is generated where by the maximum number of enemies is set by how many can physically fit on the spawn areas.The difficults scales with the level and the number of spawn points in the map.

The bug where enemies were spawning ontop of each other no longer it there , or at least it shouldn't be. Occasionally enemies get stuck at ledges but this isn't to do with this PR I don't think.

### CODE REVIEW

File: Moonshot/baddies/Baddie.gd [hunk 1]
```diff
@@ -47,7 +47,7 @@ func move(delta) -> void:
 func check_player_colision(should_damage_player = false) -> bool:
 	for i in body.get_slide_count():
 		var collision = body.get_slide_collision(i)
-		if collision.collider.name == "Player":
```
Comment (whitew1994WW) [inline Moonshot/baddies/Baddie.gd:50]: Fix that stelios suggested to an earlier bug

Comment (Jamesbarford) [reply Moonshot/baddies/Baddie.gd:50]: I'd remove this, it's tracked in a separate PR & issue

**Comment by Jamesbarford** [comment]
Is this easily modify-able @whitew1994WW ? As this we may want some post prototype tweaking predicated on how the game feels

**Comment by whitew1994WW** [comment]
Yep easy to modify difficulty of the level, we can tweak during testing.

**Comment by Jamesbarford** [comment]
Other than the point on the fix, think it looks good. @kepler471  is probably in a better position a look at the code.

**Review by kepler471 — APPROVED** [review]

**Pull request merged**